### PR TITLE
[cling] Resolve some of the test failures on osx

### PR DESCRIPTION
**Pull request by vgvassilev**

### CODE REVIEW

File: interpreter/cling/test/lit.cfg [hunk 1]
```diff
@@ -113,6 +112,11 @@ if platform.system() not in ['Windows'] or lit_config.getBashPath() != '':
 if platform.system() not in ['Windows']:
     config.available_features.add('not_system-windows')
 
+lit.util.usePlatformSdkOnDarwin(config, lit_config)
+macOSSDKVersion = lit.util.findPlatformSdkVersionOnMacOS(config, lit_config)
+if macOSSDKVersion is not None:
+    config.available_features.add('macos-sdk-' + str(macOSSDKVersion))
```
Comment (hahnjo) [inline interpreter/cling/test/lit.cfg:118]: This should not be needed, we don't use the `macos-sdk-*` features. The important call is probably `usePlatformSdkOnDarwin` which sets `SDKROOT` in the environment.

Comment (vgvassilev) [reply interpreter/cling/test/lit.cfg:118]: Agreed. However, we tend to copy parts of the clang lit.cfg and they have it. That would enable us to do XFAIL on it for example and reduce the amount of differences we have against that file..

Comment (hahnjo) [reply interpreter/cling/test/lit.cfg:118]: Why? It's a separate file, I don't understand which "differences"? We don't need it right now, and I just removed a bunch of unused lines that were leading to warnings in debbd9bb1433400bb58ae19bca7bff4b61e2a83b...

File: interpreter/cling/test/lit.cfg [hunk 2]
```diff
@@ -109,6 +108,12 @@ config.substitutions.append(('%shlibext', config.shlibext))
 if platform.system() not in ['Windows'] or lit_config.getBashPath() != '':
     config.available_features.add('shell')
 
+# Filter-out tests not supposed to run on Windows
+if platform.system() not in ['Windows']:
+    config.available_features.add('not_system-windows')
+
```
Comment (hahnjo) [inline interpreter/cling/test/lit.cfg:114]: We don't need this either

**Review by hahnjo — COMMENTED** [review]
I don't remember how the first commit fixes errors, it seems to throw errors in more cases. Can you remind me?

**Comment by vgvassilev** [comment]
> I don't remember how the first commit fixes errors, it seems to throw errors in more cases. Can you remind me?

By removing the error-unchecked assertions.

File: interpreter/cling/test/lit.cfg (hunk 2 above)
Comment (hahnjo) [reply interpreter/cling/test/lit.cfg:114]: Actually, it was removed as part of https://github.com/root-project/root/commit/e4823b01203e663d6efb1b7b1c9297b8d64f3afc, so it should not be re-introduced.

**Review by hahnjo — APPROVED** [review]
Thanks!

**Pull request merged**